operator/internal/decommissioning/statefulset_decomissioner.go

```diff
@@ -0,0 +1,753 @@
+				log.Error(err, "decommissioning broker", "broker", brokersToDecommission[0])
+				return err
+			}
+
+			return nil
+		})
+		if err != nil {
+			return false, err
+		}
+
+		if decommissioned {
+			// we decommissioned something, return immediately and wait until the process is fully complete
+			// before attempting our next decommission
+			return true, nil
+		}
+	}
+
+	// we may not have decommissioned anything above, but go ahead and requeue if we have anything
+	// in either of our caches since we have something that might need decommissioning soon
+	return s.delayedBrokerIDCache.Size(setCacheKey) > 0 || s.delayedVolumeCache.Size(setCacheKey) > 0, nil
+}
+
+// findUnboundVolumeClaims fetches any PVCs associated with the StatefulSet that aren't actively attached
+// to a pod.
+//
+// Much of this code is copied from the original decommissioner and refactored, but the basic idea
+// is:
+//
+// 1. Pull any pods matching the labels for the stateful set's pod template that are in the same namespace
+// 2. Pull any pvcs matching the labels for the stateful set's volume claim template (though the component adds a "NAME-statefulset")
+// 3. Find unbound volumes by checking that the pods we pulled reference every volume claim
+//
+// NB: this bit follows the original implementation that has a potential race-condition in the cache, where a PVC may come online
+// and be in-cache but the corresponding pod has not yet populated into the cache. In this case the PVC could be marked for deletion
+// despite the fact that it's still bound to a pod. In such a case the pvc-protection finalizer put in-place by core keeps the
+// PVC from being deleted until the pod is deleted. Due to the skip of already-deleted PVCs below, these PVCs should
+// just get GC'd when the pod is finally decommissioned. HOWEVER, this is addressed by wrapping the volume claim deletion
+// logic in an extra caching layer in the main reconciliation loop that only deletes a volume after it has been seen
+// as unbound n times with m amount of time between checks. This gives the pod and PVC time to both enter cache
+// so that the PVC will not be decommissioned while still being legitimately bound to a pod.
+func (s *StatefulSetDecomissioner) findUnboundVolumeClaims(ctx context.Context, set *appsv1.StatefulSet) ([]*corev1.PersistentVolumeClaim, error) {
+	pods := &corev1.PodList{}
+	if err := s.client.List(ctx, pods, client.InNamespace(set.Namespace), client.MatchingLabels(set.Spec.Template.Labels)); err != nil {
+		return nil, fmt.Errorf("listing pods: %w", err)
+	}
+
+	dataVolumeLabels := client.MatchingLabels{}
+	for _, template := range set.Spec.VolumeClaimTemplates {
+		if template.Name == datadirVolume {
+			dataVolumeLabels = template.Labels
+			break
+		}
+	}
+	// the first part of this, "redpanda" is the component name (i.e. redpanda, console, etc.)
+	dataVolumeLabels[k8sComponentLabelKey] = "redpanda-statefulset"
```

Comment:
I'm not sure why this is hardcoded?

Shouldn't it be https://github.com/redpanda-data/helm-charts/blob/b0a8f611127d405d0811eae052fe75d56a70799d/charts/redpanda/statefulset.go#L1186 or even it could be overwrite by user?